### PR TITLE
feat(nav): add highlighting to current url on nav bar

### PR DESCRIPTION
**Pull request by sheepdog-mcfry**
The title of the current page will be a  blue colour in the nav bar to highlight the current position of the user on the website.

### CODE REVIEW

File: layouts/default.hbs [hunk 1]
```diff
@@ -97,5 +97,14 @@
   {{/if}}
   <script src="{{ webRoot }}script/vendor.js" ></script>
   <script src="{{ webRoot }}script/script.js" ></script>
+  <script type="text/javascript">
+    window.pageReady = (container) => {
+      for (var i = 0; i < document.links.length; i++) {
```
Comment (jorunfa) [inline layouts/default.hbs]: Something like this instead?

```
document.querySelectorAll(`a.docs-nav__item[href='${window.location.pathname}']`).forEach(el => el.classList.add('active'));
```

File: script/sidebar.js [hunk 2]
```diff
@@ -6,6 +6,18 @@ function closeSidebar() {
     document.getElementById("docs-sidebar").classList.remove("docs-sidebar--open");
 }
 
+function updateNavClass(newclass) {
+    var navList = document.querySelectorAll("."+newclass);
```
Comment (jorunfa) [inline script/sidebar.js]: Maybe something like this?
```
document.querySelectorAll(`a.docs-nav__item:not([href='${window.location.pathname}'])`).forEach(el => el.classList.remove('active'));
```

File: style/layout/_nav.scss [hunk 3]
```diff
@@ -117,3 +117,8 @@
     }
   }
 }
+
+.active {
+  color: #007ace !important;
```
Comment (jorunfa) [inline style/layout/_nav.scss]: Do we have to use important here?

File: layouts/default.hbs (hunk 1 above)
Comment (jorunfa) [reply layouts/default.hbs]: Because we don't want to add it to all links, just nav links, at least.

File: style/layout/_nav.scss (hunk 3 above)
Comment (sheepdog-mcfry) [reply style/layout/_nav.scss]: removed

File: layouts/default.hbs (hunk 1 above)
Comment (sheepdog-mcfry) [reply layouts/default.hbs]: should not be a script here.

File: script/sidebar.js (hunk 2 above)
Comment (sheepdog-mcfry) [reply script/sidebar.js]: updated as per David and your suggestions

File: script/sidebar.js [hunk 4]
```diff
@@ -6,6 +6,14 @@ function closeSidebar() {
     document.getElementById("docs-sidebar").classList.remove("docs-sidebar--open");
 }
 
+function setActiveLink() {
+    document.querySelectorAll("a.docs-nav__item").forEach(el => el.classList.remove("docs-nav-active-url"));
+    const activeLink = document.querySelector(`a[href='${window.location.pathname}']`);
```
Comment (jorunfa) [inline script/sidebar.js]: `a[href...` ->
`a.docs-nav__item[href...`

File: script/sidebar.js [hunk 5]
```diff
@@ -6,6 +6,14 @@ function closeSidebar() {
     document.getElementById("docs-sidebar").classList.remove("docs-sidebar--open");
 }
 
+function setActiveLink() {
+    document.querySelectorAll("a.docs-nav__item").forEach(el => el.classList.remove("docs-nav-active-url"));
```
Comment (jorunfa) [inline script/sidebar.js]: `docs-nav__item` ->
`docs-nav-active-url` in the querySelector. Then you only have to check one element each time, basically.

Comment (jorunfa) [reply script/sidebar.js]: That is

```
document.querySelectorAll("a.docs-nav-active-url").forEach(el => el.classList.remove("docs-nav-active-url"));
```

File: script/script.js [hunk 6]
```diff
@@ -9,8 +9,10 @@ document.addEventListener("DOMContentLoaded", function(event) {
 
 function initBarba() {
     Barba.Pjax.start();
+    setActiveLink();
```
Comment (tipsy) [inline script/script.js]: Move to new line before `initBarba();` call?

Comment (sheepdog-mcfry) [reply script/script.js]: Done

File: script/sidebar.js (hunk 5 above)
Comment (sheepdog-mcfry) [reply script/sidebar.js]: Done sort of, added the new class as discussed on slack

File: script/sidebar.js (hunk 4 above)
Comment (sheepdog-mcfry) [reply script/sidebar.js]: Done

**Pull request merged**